modules/tensor_mechanics/src/materials/RecomputeGeneralReturnStressIncrement.C

```diff
@@ -0,0 +1,53 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "RecomputeGeneralReturnStressIncrement.h"
+
+template<>
+InputParameters validParams<RecomputeGeneralReturnStressIncrement>()
+{
+  InputParameters params = validParams<Material>();
+  params.addParam<unsigned int>("max_iterations", 30, "Maximum number of newton iterations in the radial return material");
+  params.addParam<std::string>("base_name", "Optional parameter that allows the user to define multiple mechanics material systems on the same block, i.e. for multiple phases");
+
+  return params;
+}
+
+
+RecomputeGeneralReturnStressIncrement::RecomputeGeneralReturnStressIncrement(const InputParameters & parameters) :
+    Material(parameters),
+    _base_name(isParamValid("base_name") ? getParam<std::string>("base_name") + "_" : "" ),
+    _return_stress_increment(declareProperty<RankTwoTensor>("return_stress_increment")),
+    _inelastic_strain_increment(declareProperty<RankTwoTensor>("inelastic_strain_increment")),
+    _max_its(parameters.get<unsigned int>("max_iterations")),
+    _elasticity_tensor(getMaterialPropertyByName<ElasticityTensorR4>(_base_name + "elasticity_tensor")),
+    _strain_increment(getMaterialProperty<RankTwoTensor>(_base_name + "strain_increment")),
+    _stress(getMaterialPropertyByName<RankTwoTensor>(_base_name + "stress")),
+    _stress_old(getMaterialPropertyOldByName<RankTwoTensor>(_base_name + "stress"))
+{
+  if (_compute)
+    mooseError("The return stress increment classes are intended to be iterative materials; add ComputeReturnMappingStress to the input file and set compute = false in " << _name );
```

Comment:
That would be a nice way to prevent the user from accidentally trying to run this material as a standalone material....unless of course @bwspenc wants these single materials to be able to run on their own.  Ben, are you okay with these plasticity (and eventually creep and damage) models being forced to always use the return mapping call as is done in solid mechanics currently?